xslt/library_patron.xsl

```diff
@@ -76,41 +76,50 @@
 			</xsl:if>
 		</xsl:variable>
 		<!--****************-->
-		<!-- Name - the registered name which is always present. Use display name first, otherwise use name type that takes up three columns. The format is first name, middle initial, first name,-->
+		<!-- Name - the display name comes from the preferred name in stanfordyou. Preferred name defaults as registered name. -->
+		<!-- First and middle can be blank but last cannot in stanfordyou, so we should grab registered first just in case. -->
 		<!--****************-->
-		<xsl:variable name="USER_NAME">
+		<xsl:variable name="USER_FIRST_NAME">
 			<xsl:choose>
-				<xsl:when test="name[@type='display']">
-					<xsl:value-of select="name/last"/>
-					<xsl:text>, </xsl:text>
-					<xsl:value-of select="name/first"/>
-					<xsl:text> </xsl:text>
-					<xsl:value-of select="name/middle"/>
-				</xsl:when>
-				<xsl:when test="@name">
-					<xsl:value-of select="@name"/>
+				<xsl:when test="name[@type='display']/first">
+					<xsl:value-of select="name[@type='display']/first"/>
 				</xsl:when>
+				<xsl:otherwise>
+					<xsl:value-of select="name[@type='registered']/first" />
+				</xsl:otherwise>
 			</xsl:choose>
 		</xsl:variable>
-		<xsl:variable name="USER_FIRST_NAME">
+		<xsl:variable name="USER_MIDDLE_NAME">
 			<xsl:choose>
-				<xsl:when test="name[@type='display']">
-					<xsl:value-of select="name/first"/>
+				<xsl:when test="name[@type='display']/middle">
+					<xsl:value-of select="name[@type='display']/middle"/>
 				</xsl:when>
```

Comment:
do we not need an otherwise block for USER_MIDDLE_NAME, like we have for USER_FIRST_NAME and USER_LAST_NAME?

Reply:
I wasn't sure since middle seemed optional. The way I wrote it, we will never get middle name if the user doesn't have anything in the preferred name field for middle name, which is probably not desired. I'll add an otherwise for middle name.

Reply:
Oh, I remember now why I did this. What if a person doesn't want their middle name to be in the preferred name? So they blank out the middle name field in stanfordyou and the XML document we get is like this:
```
<name type="registered" visibility="none">Lastname, Registeredname Y
        <first nval="firstname">Registeredname</first>
        <middle nval="y">Y</middle>
        <last nval="lastname">Lastname</last>
</name>
<name type="display" visibility="stanford">Lastname, Displayname
        <first nval="firstname">Displayname</first>
        <last nval="lastname">Lastname</last>
</name>
```
So then we end up putting "Y" in USER_NAME_MIDDLE and constructing USER_NAME to be "Lastname, Displayname Y". Okay, so I see now why we were taking `<Person name="Lastname, Displayname" >` previously to populate USER_NAME (as shown in https://github.com/sul-dlss/registry-harvester/commit/af9262b4eee4b978abb479e1f4718ddb22e19cdc).

Per conversation with Darsi, we should grab middle and last name from the display name node if anything is in the first name element, otherwise grab data from the registered name node (although it looks like the display name node will always be a duplicate of whatever is in the registered name node if a user hasn't modified preferred name in stanfordyou).